src/base.jl

```diff
@@ -44,30 +46,44 @@ struct PowerSystem{L <: ElectricLoad,
     branches::B
     storage::S
     basepower::Float64 # [MVA]
-    time_periods::Int64
+    forecasts::F
+    services::V
+    annex::Union{Nothing,Dict{Any,Any}}
 
     function PowerSystem(buses::Array{Bus,1},
                         generators::Array{G,1},
                         loads::Array{L,1},
                         branches::Nothing,
                         storage::Nothing,
-                        basepower::Float64; kwargs...) where {G <: Generator, L <: ElectricLoad}
+                        basepower::Float64,
+                        forecasts::F,
+                        services::V,
+                        annex::Union{Nothing,Dict{Any,Any}}; kwargs...) where {G <: Generator, 
+                                                                                L <: ElectricLoad,
+                                                                                V <: Union{Nothing,Array{ <: Service,1}},
+                                                                                F <: Union{Nothing,Dict{Symbol,Array{ <: Forecast,1}}}}
         
+        sources = genclassifier(generators);
         runchecks = in(:runchecks, keys(kwargs)) ? kwargs[:runchecks] : true
         if runchecks
-                generators = checkramp(generators, minimumtimestep(loads))
+                if (!isa(forecasts,Nothing)) & (length(forecasts)>0)
+                        generators = checkramp(generators, minimumtimestep(loads))
+                        time_length = timeseriescheckload(loads)
+                        !isa(sources.renewable, Nothing) ? timeserieschecksources(sources.renewable, time_length) : true
```

Comment:
In the coding standards we are trying to move away from this idiom.  You could do 
if !isa(...)
    timeserieschecksources()
end

or 
!isa(...) && timeserieschecksources()

Reply:
now there is a function `isnothing` we should use in this instances also.